src/widget/wtrackproperty.cpp

```diff
@@ -1,18 +1,59 @@
+#include "widget/wtrackproperty.h"
+
+// Qt includes
+#include <QMenu>
+#include <QtCore/QItemSelectionModel>
+#include <QtWidgets/QInputDialog>
+#include <QtWidgets/QCheckBox>
+#include <QtWidgets/QWidgetAction>
 
-#include <QDebug>
-#include <QUrl>
+// std includes
+#include <utility>
 
+// Project includes
 #include "control/controlobject.h"
-#include "widget/wtrackproperty.h"
+#include <library/crate/cratefeaturehelper.h>
+#include "library/dao/playlistdao.h"
+#include "library/trackcollection.h"
+#include "library/trackcollectionmanager.h"
+#include "library/trackmodel.h"
+#include "track/track.h"
 #include "util/dnd.h"
+#include "util/desktophelper.h"
 
 WTrackProperty::WTrackProperty(const char* group,
                                UserSettingsPointer pConfig,
-                               QWidget* pParent)
+                               QWidget* pParent,
+                               TrackCollectionManager* pTrackCollectionManager)
         : WLabel(pParent),
           m_pGroup(group),
-          m_pConfig(pConfig) {
+          m_pConfig(std::move(pConfig)),
+          m_pTrackCollectionManager(pTrackCollectionManager),
+          m_bPlaylistMenuLoaded(false),
+          m_bCrateMenuLoaded(false) {
     setAcceptDrops(true);
+
+    // Setup context menu
+    m_pMenu = new QMenu(this);
+    m_pPlaylistMenu = new QMenu(this);
+    m_pPlaylistMenu->setTitle(tr("Add to Playlist"));
+    connect(m_pPlaylistMenu, SIGNAL(aboutToShow()),
+            this, SLOT(slotPopulatePlaylistMenu()));
```

Comment:
Use Qt5 functor signal-slot syntax for new code. When you are editing old code, please replace the old syntax with the new (you don't need to do that for the whole file, just code that you edit).

https://wiki.qt.io/New_Signal_Slot_Syntax

Reply:
Sure. I'll upgrade the code I touch to the new syntax.